### PR TITLE
graphic: use GLVND to dispatch graphic API calls

### PR DESCRIPTION
**Pull request by 5schatten**
This PR adds https://github.com/NVIDIA/libglvnd which is already supported by MESA 3D and newer Nvidia drivers. It slightly updates the `xorg-configure` script to create matching symlinks for the new libs and adjusts the mesa & xf86-video-nvidia package to use libglvnd. These changes are mandatory for all upcoming nvidia drivers past version 430 which will drop non-GLVND binaries.

It was tested on my Intel i3-6100, Asrock J3455-ITX systems which use the IGP, my workstation which uses a GTX 1060 6GB, on a systems using the legacy-driver with a [8400GS](https://forum.libreelec.tv/thread/12662-libreelec-rr-9-x-emulationstation-retroarch-pegasus-dolphinqt-moonlight-chrome-s/?postID=126938#post126938) and on a ION2 based system owned by @MilhouseVH 

Changes:
- libglvnd: initial package v1.1.1 / added pkgconfig patch
- mesa: updated to support libglvnd / clean up
- xf86-video-nvidia: updated to support libglvnd / clean up / updated lima & panfrost to 19.2.0-rc1 / enabled GLES v1 for OpenGL ES builds -> necessary for some third party addons which needs these libs & headers
- xorg-server: updated to support libglvnd / clean up
- libepoxy: switched toolchain to meson / clean up

some docs:
https://devtalk.nvidia.com/default/topic/915640/multiple-glx-client-libraries-in-the-nvidia-linux-driver-installer-package/
https://devtalk.nvidia.com/default/topic/1032650/linux/unix-graphics-feature-deprecation-schedule/

### CODE REVIEW

**Comment by 5schatten** [comment]
Latest changes https://github.com/NVIDIA/libglvnd/pull/179 build and work fine with mesa `19.2.0-rc1` :thinking: @MilhouseVH mesa `19.1.5` works fine with this update too?

**Comment by lrusak** [comment]
@evelikov if you possibly have some free time would you be able to give this a quick review? I know you're the one that said we should switch to use GLVND some time ago. 👍

File: packages/graphics/libepoxy/package.mk [hunk 1]
```diff
@@ -12,19 +12,18 @@ PKG_VERSION="1.5.3"
...
-if [ "$OPENGLES" != "no" ]; then
-  PKG_DEPENDS_TARGET+=" $OPENGLES"
+if [ "${OPENGLES_SUPPORT}" = "yes" ]; then
+  PKG_DEPENDS_TARGET+=" ${OPENGLES}"
 fi
 
-if [ "$DISPLAYSERVER" != "x11" ]; then
-  PKG_CONFIGURE_OPTS_TARGET="--disable-glx"
+if [ "${DISPLAYSERVER}" != "x11" ]; then
+  PKG_MESON_OPTS_TARGET="-D glx=no"
```
Comment (lrusak) [inline packages/graphics/libepoxy/package.mk]: no space between `-D glx`

File: packages/x11/driver/xf86-video-nvidia/package.mk [hunk 2]
```diff
@@ -11,63 +11,62 @@ PKG_SHA256="19ff49073e88245666825722e2c814d369ab4d01b2883a954ce0bfbe9c3b22ab"
...
+    ${STRIP} --strip-debug nvidia.ko
   cd ..
 }
 
 makeinstall_target() {
-  mkdir -p $INSTALL/$XORG_PATH_MODULES/drivers
-    cp -P nvidia_drv.so $INSTALL/$XORG_PATH_MODULES/drivers/nvidia-main_drv.so
-    ln -sf /var/lib/nvidia_drv.so $INSTALL/$XORG_PATH_MODULES/drivers/nvidia_drv.so
+  mkdir -p ${INSTALL}/${XORG_PATH_MODULES}/drivers
+    cp -P nvidia_drv.so           ${INSTALL}/${XORG_PATH_MODULES}/drivers/nvidia-main_drv.so
```
Comment (lrusak) [inline packages/x11/driver/xf86-video-nvidia/package.mk]: don't align things like this. Same for all the lines below

File: packages/x11/xserver/xorg-server/scripts/xorg-configure [hunk 3]
```diff
@@ -55,39 +56,43 @@ logger -t Xorg "### creating start options ###"
...
+    ln -sf /usr/lib/libGL_glvnd.so.1                         /var/lib/libGL.so
     ln -sf /usr/lib/xorg/modules/extensions/libglx_nvidia.so /var/lib/libglx.so
-    ln -sf /usr/lib/libnvidia-ml.so.@NVIDIA_VERSION@ /var/lib/libnvidia-ml.so.1
-    ln -sf /usr/lib/xorg/modules/drivers/nvidia-main_drv.so /var/lib/nvidia_drv.so
-    ln -sf /usr/lib/nvidia/nvidia.ko /var/lib/nvidia.ko
-    ln -sf /usr/lib/vdpau/libvdpau_nvidia-main.so.1 /var/lib/libvdpau_nvidia.so.1
-    ln -sf /usr/lib/vdpau/libvdpau_nvidia-main.so.1 /var/lib/libvdpau_nvidia.so
-    ln -sf /usr/bin/nvidia-main-smi /var/lib/nvidia-smi
-    ln -sf /usr/bin/nvidia-main-xconfig /var/lib/nvidia-xconfig
+    ln -sf /usr/lib/libnvidia-ml.so.@NVIDIA_VERSION@         /var/lib/libnvidia-ml.so.1
```
Comment (lrusak) [inline packages/x11/xserver/xorg-server/scripts/xorg-configure]: don't align like this

**Comment by lrusak** [comment]
looks like someone has a lot of cleanup to do now that this is merged 😸

**Comment by 5schatten** [comment]
Well what's the problem with the alignment? Otherwise readability is down to zero if you install a bunch of files :thinking:

**Comment by lrusak** [comment]
because if you add or remove one then you may have to move the alignment for all the lines.

**Comment by 5schatten** [comment]
That's the whole point. Nvidia tends to add, remove or rename libs out of the blue and with this alignment I see on first sight which is missing or has changed. If the "price" for this is to push the spacebar _if_ the file with longest name has randomly changed then I'm willing to do it. `¯\_(ツ)_/¯`

Beside that I guess nobody else is maintaining the Nvidia stuff actively anyway :wink:

**Comment by lrusak** [comment]
we have never used alignment before and we aren't about to start now. So please fix it or revert the whole lot. Thanks.

**Pull request merged**